### ISSUE TITLE
Gradients not tracked through evaluation_from_torch

### ISSUE THREAD

**Issue by bitanath** [Kent0n-Li/SAMScore#2]
Hi, 

I noticed gradients are not tracked through evaluation_from_torch. Is this expected behavior?

```
tensor1.requires_grad = True
tensor2.requires_grad = True

SAMScore_Evaluation = samscore.SAMScore(model_type = "vit_b" )
samscore_result = SAMScore_Evaluation.evaluation_from_torch(source=tensor1.unsqueeze(0),generated=tensor2.unsqueeze(0))
```

Samscore result is `tensor([0.8888])` without the gradients

Am I doing something wrong here?

**Comment by bitanath** [comment]
Okay just read through the code -> gradients are disabled in the forward pass through the sam model.. but why? Embeddings can be computed differentiably, and SAM score is just the average of cosine similarities of the embeddings? Hoping for some clarity on the matter.

```
with torch.no_grad():
            embedding = sam_model.image_encoder(resampled_image)
```
